polkadot-protocol/protocol-components/accounts/address-formats.md

```diff
@@ -0,0 +1,215 @@
+
+The advantage of using additional checksum bytes lies in the increased protection they offer against input errors and index alterations. However, this comes at the expense of expanding the textual address by a few extra characters. 
+
+In the case of the account ID format, this increase is negligible, which is why no 1-byte alternative is provided. For shorter account-index formats, the extra byte constitutes a significantly larger portion of the final address. 
+
+Therefore, it is left to higher levels of the stack (rather than the end user) to determine the best trade-off based on their specific needs.
+
+## Address Types and Network Registry
+
+The [SS58 registry](https://github.com/paritytech/ss58-registry) is the canonical listing of all address type identifiers and how they map to Substrate-based networks.
+
+## Encoding Address and Network Identifiers
+
+Identifiers up to the value of 64 can be represented using a simple address format, where the least significant byte of the network identifier is encoded as the first byte of the address.
+
+For identifiers ranging from `64` to `16,383`, the full address format must be utilized.
+
+The full address encoding requires special handling because SCALE encoding as little endian requires the first two bits to be used for the `01` prefix. 
+
+To encode the network identifier, the full address format treats the first two bytes as a `16-bit` sequence, and disregards the first two bits of that sequence to account for the `01` prefix. 
+
+The remaining 14 bits encode the network identifier value as little endian, with the assumption that the two missing higher order bits are zero. This effectively spreads the low-order byte across the boundary between the two bytes.
+
+For example, the 14-bit identifier `0b00HHHHHH_MMLLLLLL` is expressed in two bytes as:
+
+```
+0b01LLLLLL
+0bHHHHHHMM
+```
+!!! note 
+    Identifiers of `16384` and beyond aren't currently supported.
+
+## Validating Addresses
+
+You can verify that a value is a valid `SS58` address by using the subkey `inspect command` or though the `Polkadot-JS API`.
+
+### Using subkey
+
+The basic syntax for the `subkey inspect` command is:
+
+```
+subkey inspect [flags] [options] uri
+```
+
+For the `uri` command-line argument, you can specify the secret seed phrase, a hex-encoded private key, or an `SS58` address. If the input is a valid address, the `subkey` program displays the corresponding hex-encoded public key, account identifier, and SS58 addresses. 
+
+For example, to inspect the public keys derived from a secret seed phrase, you can run a command similar to the following:
+
+```bash
+subkey inspect "caution juice atom organ advance problem want pledge someone senior holiday very"
+```
+The command displays output similar to the following:
+
+```
+Secret phrase `caution juice atom organ advance problem want pledge someone senior holiday very` is account:
```

Comment:
Use termynal plugin to display terminal output

Reply:
what's the name of the plug in?

Reply:
You have to wrap the code with something similar to the following example:
https://raw.githubusercontent.com/polkadot-developers/polkadot-docs/refs/heads/master/.snippets/code/develop/application-devs/tooling/asset-transfer-api/overview/paraToPara.md